shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -811,25 +730,8 @@ bool PlatformView::HandleFlutterPlatformViewsChannelPlatformMessage(
                 std::vector<uint8_t>({'[', '0', ']'})));
           }
         });
-    auto on_view_bound =
-        [weak = weak_factory_.GetWeakPtr(),
-         platform_task_runner = task_runners_.GetPlatformTaskRunner(),
-         view_id = view_id_raw](scenic::ResourceId resource_id) {
-          platform_task_runner->PostTask([weak, view_id, resource_id]() {
-            if (!weak) {
-              FML_LOG(WARNING)
-                  << "ViewHolder bound to PlatformView after PlatformView was "
-                     "destroyed; ignoring.";
-              return;
-            }
-
-            FML_DCHECK(weak->child_view_ids_.count(resource_id) == 0);
-            weak->child_view_ids_[resource_id] = view_id;
-          });
-        };
-    on_create_view_callback_(
-        view_id_raw, std::move(on_view_created), std::move(on_view_bound),
-        hit_testable->value.GetBool(), focusable->value.GetBool());
+    OnCreateView(std::move(on_view_created), view_id->value.GetUint64(),
+                 hit_testable->value.GetBool(), focusable->value.GetBool());
```

Comment:
Sorry, I know this isn't related to your change, but to make dereferencing hit_testable, view_id, focusable etc. safe, do we also need to check that the result of args.FindMember != root.MemberEnd()?

Reply:
That's a good question.  I don't know rapidjson well enough to answer confidently.  We could find out the answer by trying to send one of these platform messages, but with one of the fields edited out.

If you have time, would you run that experiment and file a bug if it results in undefined behavior?